### ISSUE TITLE
Connection options

### ISSUE THREAD

**Issue by pdewouters** [ifsnop/mysqldump-php#84]
I can't seem to find a way to set this flag:

````
--protocol={TCP|SOCKET|PIPE|MEMORY}

The connection protocol to use for connecting to the server. It is useful when the other connection parameters normally would cause a protocol to be used other than the one you want. For details on the permissible values, see Section 4.2.2, “Connecting to the MySQL Server”.
````

What if I am connecting via a socket?

For example:
https://help.1and1.com/hosting-c37630/databases-c85147/mysql-database-c37730/connect-to-a-mysql5-database-on-a-1and1-managed-server-a752513.html\

Or will the library correctly parse the host value?

**Comment by ifsnop** [comment]
tl;dr Not yet, give me a couple of days to thing best way to do it.

Although the link you provide doesn't work, I get the idea. The --protocol option is only useful to force a different type of connection, for example, connecting to localhost by default makes the connection using a socket. If you want to connect using tcp, you then use --protocol=TCP.

Please see:
http://php.net/manual/en/ref.pdo-mysql.connection.php

~~~
When the host name is set to "localhost", then the connection to the server is made thru a domain socket.
~~~

mysqldmp-php passes the host option directly to the dsn option of the constructor.

https://github.com/ifsnop/mysqldump-php/blob/master/src/Ifsnop/Mysqldump/Mysqldump.php#L194

Unfortunately, there is no way to connect using a socket different from the default one with the current mysqldump code.

Changes needed to support connecting through a socket would probably made the next version of mysqldump-php incompatible with the current one. Last option is currently not supported.
~~~
mysql:host=localhost;dbname=testdb
mysql:host=localhost;port=3307;dbname=testdb
mysql:unix_socket=/tmp/mysql.sock;dbname=testdb
~~~

**Comment by pdewouters** [comment]
Ah, you're right - it was working when I posted. Anyway, the use case would be when the hostname is followed by a custom path : `localhost:/tmp/mysql5`, so yes like in your example :)

**Comment by ifsnop** [comment]
Looks like using a DSN string is the best way to handle connection strings to databases. In the end, PDO will take care of it without parsing.

**Comment by pdewouters** [comment]
makes sense

**Comment by ifsnop** [comment]
Please, take a look at the devel branch.

**Comment by ifsnop** [comment]
Everything should be fixed by now.

I will bump version to 2.0 because now the constructor takes a dsn string as a parameter, the same that is needed by PDO to create the connection, and everything is parsed from there. Also now the default filename name equals to stdout, to mimic mysqldump behavoiur. 

I rewrote some parts of old code, but didn't tag it yet to 2.0. If more positive comments are made, I will be very happy to improve things.
